packages/web-scripts/src/allSettled.ts

```diff
@@ -0,0 +1,36 @@
+interface PromiseResult {
+  status: string;
+  reason?: Error;
+  value?: any;
+}
+interface PromiseRejection extends PromiseResult {
+  status: 'rejected';
+  reason: Error;
+}
+interface PromiseFulfillment<T> extends PromiseResult {
+  status: 'fulfilled';
+  value: T;
+}
+
+export function allSettled<T>(arr: Promise<T>[]): Promise<PromiseResult[]> {
```

Comment:
Another, possibly more idiomatic, way to accomplish these types is:

```ts
type PromiseRejection {
  status: 'rejected';
  reason: Error;
}
type PromiseFulfillment<T> {
  status: 'fulfilled';
  value: T;
}

type PromiseResult = PromiseRejection | PromiseFulfillment;
```

Later, when you use this function, you can even use a `switch` statement or `if` and get autocomplete for valid `status`!

Reply:
YES this is better.

Reply:
the `allSettled` module ultimately did not end up being necessary, but I'm glad I learned this.